### PR TITLE
Cleanup Battle Names (Bruno2, Hop2, Circhester2, etc.)

### LINKED ISSUE #934: [BUG] Labeling/Spelling/General wording issues
The usage of Pokemon vs Pok**é**mon is not consistent. 

Some example locations:
1. The new Pokeball selector modal (Main header and description)
2. Hatchery Tab name.
3. Sorting method name ("Pokemon #")
4. Underground Help section
5. Pokeball shop entry.
6. Damage calculator modal
7. Statistics Modal entries
8. Couple of towns/Dungeons (Pokemon Tower, Pokemon Mansion, the Leagues).

Not sure if anyone wants to go through the effort to change all of these (I'm not sure if I recorded all of them). In addition, changing the dungeon names would also mess with how we store the save data, which doesn't sound fun.

We also use Bruno2 for the name of Bruno in Johto. I saw at least one person confused about that. 

In general, the item names are a bit off.
1. Most item names have their second word not capitalized. This is due to how we obtain the name by just replacing underscores with spaces.
2. The Restores don't even have spaces.

Cleaned up List:
- [x] Pokemon/Pokeball references
- [x] Item names
- [x] Updating names for Towns (Pokemon Tower, Pokemon Mansion, Pokemon Day Care, Pokemon Leagues)
- [ ] Updating certain trainer battles (Bruno2)

### LINKED ISSUE #2920: [BUG] Alola Trials are named Gyms
**Info**
 All versions

**Description:**
Right now, all trials in Alola displays gyms in the label of the button to start the fight. It is specifically specified that there are no gyms in Alola (see screenshot)

**Steps to Reproduce:**
Go to Alola
Check any "gyms"

**Expected Behavior:**
The trial challenge button should display Trial and not Gym

**Screenshots:**
image

### PR DESCRIPTION
**Pull request by CorgiOnNeptune**
Should clean up all cases of the display values for gym leader/town names in the achievements and quests. 

Also changed the Alola 'gyms' to the would-be canon trial names.

Closes #934, closes #2920

### CODE REVIEW

File: src/modules/requirements/TemporaryBattleRequirement.ts [hunk 1]
```diff
@@ -11,6 +11,6 @@ export default class TemporaryBattleRequirement extends Requirement {
     }
 
     public hint(): string {
-        return `Requires beating ${this.battleName}.`;
+        return `Requires beating ${this.battleName.replace(/\s(?=[\d])\d/g, '')}.`;
```
Comment (Jaaslet) [inline src/modules/requirements/TemporaryBattleRequirement.ts:14]: Temp battles have a display name. I know this is in modules and temp battles are not, so it's not as easy. But this seems like a temp solution

File: src/scripts/achievements/AchievementHandler.ts [hunk 2]
```diff
@@ -391,13 +391,21 @@ class AchievementHandler {
...
          * REGIONAL
          */
-        const addGymAchievements = (gyms: string[], category: GameConstants.Region | GameConstants.ExtraAchievementCategories) => {
+        const addGymAchievements = (gyms: string[], category: GameConstants.Region | GameConstants.ExtraAchievementCategories, subregion?: string) => {
             gyms.forEach(gym => {
-                const gymTitle: string = gym.includes('Elite') || gym.includes('Champion') ? gym : `${gym} Gym`;
+                const displayName = GymList[gym]?.displayName;
+
+                // Name of person's title if elite/champion, else the gym's town name
+                const gymTitle: string = gym.includes('Elite') || gym.includes('Champion') || gym.includes('Gym') ? gym.replace(/\d/g, '') : `${gym.replace(/\d/g, '')} Gym`;
```
Comment (Jaaslet) [inline src/scripts/achievements/AchievementHandler.ts:399]: This solution is most likely breaking if the player tries to track for example a Bruno achievement. We will have 2 achievements with the same name, and then we don't know which one to track, when we open a save

File: src/scripts/quests/questTypes/DefeatGymQuest.ts [hunk 3]
```diff
@@ -39,10 +39,15 @@ class DefeatGymQuest extends Quest implements QuestInterface {
...
+        const trial = GymList[this.gymTown]?.displayName?.includes('Trial');
         const desc = [];
-        desc.push(`Defeat ${this.gymTown}`);
-        if (!this.gymTown.includes('Elite') && !this.gymTown.includes('Champion')) {
-            desc.push('gym');
+
+        if (this.gymTown.includes('Elite') || this.gymTown.includes('Champion') || this.gymTown.includes('Gym')) {
+            desc.push(`Defeat ${townName}`);
+        } else {
+            desc.push(`Defeat ${leaderName}'s ${trial ? 'trial' : 'gym'} at ${townName}`);
```
Comment (Jaaslet) [inline src/scripts/quests/questTypes/DefeatGymQuest.ts:50]: Can we also use displayName here? Or add an other displayName for this situation?

File: src/scripts/gym/Gym.ts [hunk 4]
```diff
@@ -70,13 +70,16 @@ class Gym extends TownContent {
         {
             quest = true,
             achievement = true,
-        }: gymFlags = {}
+        }: gymFlags = {},
+        public displayName?: string
```
Comment (Jaaslet) [inline src/scripts/gym/Gym.ts:74]: Can we use the displayName for all the `...2` gym leaders? Then we can remove all the `.replace(/\d/g, '')` code we have

**Review by Jaaslet — CHANGES_REQUESTED** [review]
It looks good. But still needs some stuff:

File: src/modules/requirements/TemporaryBattleRequirement.ts (hunk 1 above)
Comment (CorgiOnNeptune) [reply src/modules/requirements/TemporaryBattleRequirement.ts:14]: It's definitely a temp solution haha, I have no idea how to reference the displayName from scripts here...

Comment (Jaaslet) [reply src/modules/requirements/TemporaryBattleRequirement.ts:14]: We will leave this like this for now...

**Review by Jaaslet — APPROVED** [review]
Looks good. Thanks for the PR

**Pull request merged**